### PR TITLE
[BEAM-1928] Add CombineTranslator

### PR DESCRIPTION
**Pull request by tgroh**
This translates Combines to CombinePayloads and back

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CombineTranslator.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * Methods for translating between {@link Combine.PerKey} {@link PTransform PTransforms} and {@link
+ * RunnerApi.CombinePayload} protos.
+ */
+public class CombineTranslator {
+  private static final String JAVA_SERIALIZED_COMBINE_FN_URN = "urn:beam:java:combinefn:v1";
+
+  public static CombinePayload toProto(
+      AppliedPTransform<?, ?, Combine.PerKey<?, ?, ?>> combine, SdkComponents sdkComponents)
+      throws IOException {
+    Coder<?> accumulatorCoder = null;
```
Comment (kennknowles) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CombineTranslator.java]: You'll actually need this coder. I added a hack and hit some other snags like unequal `CountFn`.

Comment (tgroh) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CombineTranslator.java]: Yeah, there were a few things, they should have been fixed. PTAL

**Comment by tgroh** [comment]
PTAL

**Pull request merged**